### ISSUE TITLE
Tokenizing URLs

### ISSUE THREAD

**Issue by dlukes** [ufal/morphodita#4]
All tokenizers bundled with MorphoDiTa split URLs at the first occurrence of a character outside the ASCII range:

```sh
$ for tok in czech english generic; do echo 'http://cs.wikipedia.org/wiki/Islámské_bankovnictví' | run_tokenizer --tokenizer=$tok --output=vertical; done
http://cs.wikipedia.org/wiki/Isl
ámské
_
bankovnictví

http://cs.wikipedia.org/wiki/Isl
ámské
_
bankovnictví

http://cs.wikipedia.org/wiki/Isl
ámské
_
bankovnictví
```

Nowadays though, URLs often contain a wide range of Unicode characters (as shown above). Would you please consider altering the behavior of the tokenizers in this respect? :)

P.S. I've checked to see whether this is somehow affected by locale settings and it doesn't seem to be.

**Comment by choroba** [comment]
When I copy the URL from a browser window (Firefox, Chrome) and paste it, I get

    https://cs.wikipedia.org/wiki/Isl%C3%A1msk%C3%A9_bankovnictv%C3%AD

**Comment by dlukes** [comment]
That's true, but still, since browsers are able to handle the unescaped form as well, these can be expected to occasionally crop up in the data (that's how I stumbled upon the example shown above)... So would you recommend a pre-processing step rather than adjusting this on MorphoDiTa's side? I guess that would be an option as well, unless users of the resulting corpus might possibly be interested specifically in how URLs are written (accented characters vs. escape codes).

**Comment by foxik** [comment]
I know about the issue and it will be fixed. Currently the tokenizer uses RFC rules (which specify that non-ASCII characters are to be encoded using the %xx escapes), but nowadays it is normal to allow non-ascii characters in URL.

The code is already in the repository, but it is not allowed by default (because according to API stability, old models have to use old tokenizer). I will enable the new tokenizers in run_tokenize and when 2.0 comes out, all new models will use it as well.

**Comment by foxik** [comment]
Fixed in f4d691a.

Note that updated tokenizers are currently only used by run_tokenizer -- existing models still use the original tokenizer (so that they behave as in older versions of MorphoDiTa). New models created by MorphoDiTa 2.0 will use the new tokenizers too (but in current development version only newly created English models use the new tokenizers, I have not yet created updated versions of Czech, Generic and External morphologies).

**Comment by dlukes** [comment]
Great, thanks a lot! :)
